Product/Production/Services/AuditRepositoryCore/src/test/java/gov/hhs/fha/nhinc/audit/transform/AuditTransformTest.java

```diff
@@ -0,0 +1,224 @@
+import gov.hhs.fha.nhinc.audit.AuditTransformConstants;
+import gov.hhs.fha.nhinc.audit.AuditTransformDataBuilder;
+import gov.hhs.fha.nhinc.common.auditlog.LogEventRequestType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import java.lang.management.ManagementFactory;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.Properties;
+import org.junit.After;
+import org.junit.AfterClass;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+
+/**
+ *
+ * @author achidamb
+ */
+public abstract class AuditTransformTest<T,K> {
+
+    public AuditTransformTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() {
+    }
+
+    @AfterClass
+    public static void tearDownClass() {
+    }
+
+    @Before
+    public void setUp() {
+    }
+
+    @After
+    public void tearDown() {
+    }
+
+    /**
+     * Test of transformMsgToAuditMsg method, of class AuditTransform.
+     */
+    /*@Test
+     public void testTransformMsgToAuditMsg() {
+     System.out.println("transformMsgToAuditMsg");
+     Object request = null;
+     AssertionType assertion = null;
+     NhinTargetSystemType target = null;
+     String direction = "";
+     String _interface = "";
+     boolean isRequesting = false;
+     Properties webContextProperties = null;
+     String serviceName = "";
+     AuditTransform instance = new AuditTransformImpl();
+     LogEventRequestType expResult = null;
+     LogEventRequestType result = instance.transformMsgToAuditMsg(request, assertion, target, direction, _interface, isRequesting, webContextProperties, serviceName);
+     assertEquals(expResult, result);
+     // TODO review the generated test code and remove the default call to fail.
+     fail("The test case is a prototype.");
+     }*/
+    /**
+     * Test of getParticipantObjectIdentification method, of class AuditTransform.
+     */
+    /*@Test
+     public void testGetParticipantObjectIdentification() {
+     System.out.println("getParticipantObjectIdentification");
+     Object request = null;
+     AssertionType assertion = null;
+     Boolean isRequesting = null;
+     AuditMessageType auditMsg = null;
+     AuditTransform instance = new AuditTransformImpl();
+     AuditMessageType expResult = null;
+     AuditMessageType result = instance.getParticipantObjectIdentification(request, assertion, isRequesting, auditMsg);
+     assertEquals(expResult, result);
+     // TODO review the generated test code and remove the default call to fail.
+     fail("The test case is a prototype.");
+     }*/
+
+    /**
+     * Test of createEventIdentification method, of class AuditTransform.
+     *
+     * @param request
+     * @param serviceName
+     * @param isRequesting
+     */
+    public void testGetEventIdentificationType(LogEventRequestType request, String serviceName, Boolean isRequesting) {
```

Comment:
I think all the methods in this class should probably be protected.  I think they are only meant to be accessed by sub classes.